misc/test/performance_test.go

```diff
@@ -3,86 +3,176 @@
 package test
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
 	"path"
+	"path/filepath"
+	"strings"
 	"testing"
 	"time"
 
+	"github.com/pulumi/pulumi-trace-tool/traces"
 	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestAccAwsGoS3Folder(t *testing.T) {
-	test := getAWSBase(t).
-		With(integration.ProgramTestOptions{
-			Dir: path.Join(getCwd(t), "..", "..", "aws-go-s3-folder"),
-			ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
-				maxWait := 10 * time.Minute
-				endpoint := stack.Outputs["websiteUrl"].(string)
-				assertHTTPResultWithRetry(t, endpoint, nil, maxWait, func(body string) bool {
-					return assert.Contains(t, body, "Hello, world!")
-				})
-			},
-		})
+type bench struct {
+	name  string
+	cloud string
+	lang  string
+}
 
+func TestAccAwsGoS3Folder(t *testing.T) {
+	benchmark := bench{"aws-go-s3-folder", "aws", "go"}
+	opts := integration.ProgramTestOptions{
+		Dir: path.Join(getCwd(t), "..", "..", benchmark.name),
+		ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+			maxWait := 10 * time.Minute
+			endpoint := stack.Outputs["websiteUrl"].(string)
+			assertHTTPResultWithRetry(t, endpoint, nil, maxWait, func(body string) bool {
+				return assert.Contains(t, body, "Hello, world!")
+			})
+		},
+	}
+	test := getBase(t, benchmark, opts)
 	integration.ProgramTest(t, &test)
+	computeMetricsInTest(t)
```

Comment:
We can't really do this as this is the set of tests we run for ALL of our acceptance tests - 

maybe we should create a specific test for your performance test - it's a copy of what is there but named differently e,g, TestAccAwsGoS3FolderBenchmark

and we remove the build tag all in the file

Reply:
Hmm yeah it did run a bit aggressively. Let me try to move it to "TestMain/teardown", and note it only will run if PULUMI_TRACING_DIR is set, that's only in the perf context. It should not run in our normal acceptance tests I don't think. Let me test it in GHA.